Add codegen-tests.sh to update or run the tests (#2799)
iovisor/bpftrace: tests/README.md

second argument. The `NAME` macro holds the test name,  which is `call_avg` in
this case.

> These tests run as part of the normal suite of unit tests unless you are not
> running LLVM 12, in which case you need to install 'nix' and run these via this

Reviewer: nit: easier to read without double negative
```suggestion
These tests run as part of the normal suite of unit tests if you are
running LLVM 12. If not,  you need to install 'nix' and run these via this
```

`./tests/codegen-tests.sh`.

#### Updating

**LLVM 12**

> If you are running LLVM 12 or want to only update specifc tests with `--gtest_filter`

Reviewer 2: Typo:
```suggestion
If you are running LLVM 12 or want to only update specific tests with `--gtest_filter`
```

Author: Good catch. I gotta run the spell checker more often.

> If you are running LLVM 12 or if you need more control over which tests are updated,

Reviewer: Unclear what "need more control" means here. I'm guessing you mean you can combine with `--gtest_filter` to only update a subset of codegen tests?

Author: Yup, updated the language to make it clearer.
